src/sentry/api/endpoints/organization_member/index.py

```diff
@@ -371,27 +375,39 @@ def post(self, request: Request, organization) -> Response:
                 status=400,
             )
 
-        with transaction.atomic(router.db_for_write(OrganizationMember)):
-            # remove any invitation requests for this email before inviting
-            existing_invite = OrganizationMember.objects.filter(
-                Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
-                | Q(invite_status=InviteStatus.REQUESTED_TO_JOIN.value),
-                email=result["email"],
-                organization=organization,
-            )
-            for om in existing_invite:
-                om.delete()
-
-            om = OrganizationMember(
-                organization=organization,
-                email=result["email"],
-                role=result["role"],
-                inviter_id=request.user.id,
-            )
+        if is_active_staff(request):
+            with transaction.atomic(router.db_for_write(OrganizationMember)):
+                om = OrganizationMember.objects.create(
+                    organization=organization,
+                    user_email=result["email"],
+                    user_id=result["user_id"],
+                    role=result["role"],
+                    inviter_id=request.user.id,
+                    invite_status=InviteStatus.APPROVED.value,
+                )
+                om.save()
+        else:
+            with transaction.atomic(router.db_for_write(OrganizationMember)):
+                # remove any invitation requests for this email before inviting
+                existing_invite = OrganizationMember.objects.filter(
+                    Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
+                    | Q(invite_status=InviteStatus.REQUESTED_TO_JOIN.value),
+                    email=result["email"],
+                    organization=organization,
+                )
+                for om in existing_invite:
+                    om.delete()
```

Comment:
nit: I know you're just moving this code, but just to be safe, i'd use another variable name here, since we're trying to say that `om` is a nice instance that we're creating in the previous if statement, and again after this for loop. Maybe just `ei`?